### PR TITLE
Fix RGB Albumentations order

### PR DESCRIPTION
**Pull request by glenn-jocher**
Second attempt at resolving RGB Albmentations order issue https://github.com/ultralytics/yolov5/pull/8695. 

First PR fix #8695 failed and reverted.

## 🛠️ PR Summary

<sub>Made with ❤️ by [Ultralytics Actions](https://github.com/ultralytics/actions)<sub>

### 🌟 Summary
Improved color handling in image augmentations and loading by ensuring consistent RGB color space usage.

### 📊 Key Changes
- Changed `cv2.COLOR_BGR2HSV` to `cv2.COLOR_RGB2HSV` in `augment_hsv()` to start with RGB images for HSV conversion.
- Updated `cv2.COLOR_HSV2BGR` to `cv2.COLOR_HSV2RGB` after HSV operations to maintain images in RGB format.
- Removed reverse channel ordering (BGR to RGB conversion) from `transpose` operation in `__getitem__()` in `dataloaders.py`.
- Added explicit RGB format notation in `load_image()` and `cache_images_to_disk` to indicate that images are kept in RGB.

### 🎯 Purpose & Impact
- Ensures image data is consistently in RGB format throughout processing, which aligns with commonly used image formats in machine learning and reduces confusion.
- These changes should make the codebase more intuitive and reduce potential errors related to color space mismanagement, resulting in a smoother development and training experience for users working with the YOLOv5 code. 🏗️🌈

### CODE REVIEW

**Comment by glenn-jocher** [comment]
Seems like it's working. The only thing I'm worried about is non-contiguous images being passed around the dataloader. I'll check to see if they're contiguous

File: utils/dataloaders.py [hunk 1]
```diff
@@ -679,8 +679,8 @@ def load_image(self, i):
             if r != 1:  # if sizes are not equal
                 interp = cv2.INTER_LINEAR if (self.augment or r > 1) else cv2.INTER_AREA
                 im = cv2.resize(im, (int(w0 * r), int(h0 * r)), interpolation=interp)
-            return im, (h0, w0), im.shape[:2]  # im, hw_original, hw_resized
-        return self.ims[i], self.im_hw0[i], self.im_hw[i]  # im, hw_original, hw_resized
+            return im[..., ::-1], (h0, w0), im.shape[:2]  # im RGB, hw_original, hw_resized
+        return self.ims[i], self.im_hw0[i], self.im_hw[i]  # im RGB, hw_original, hw_resized
```
Comment (UnglvKitDe) [inline utils/dataloaders.py:683]: Typo: change "im" to "ims" in comment ;)

**Review by UnglvKitDe — APPROVED** [review]
Looks good :)

**Comment by UnglvKitDe** [comment]
> Seems like it's working. The only thing I'm worried about is non-contiguous images being passed around the dataloader. I'll check to see if they're contiguous

@glenn-jocher But what does that have to do with the changes here? Or do you mean they could become noncontinous because of the changes?

**Comment by glenn-jocher** [comment]
@UnglvKitDe ops on non-contiguous arrays will still run but they are no longer optimized and the operations may be much slower, i.e.:

```python
import cv2
import numpy

im_bgr = cv2.imread('data/images/bus.jpg')
im_rgb1 = im_bgr[...,::-1]
im_rgb2 = cv2.cvtColor(im_bgr, cv2.COLOR_BGR2RGB)

print(im_bgr.data.contiguous)  # True
print(im_rgb1.data.contiguous)  # False
print(im_rgb2.data.contiguous)  # True
```

**Comment by UnglvKitDe** [comment]
> @UnglvKitDe ops on non-contiguous arrays will still run but they are no longer optimized and the operations may be much slower, i.e.:
> 
> ```python
> import cv2
> import numpy
> 
> im_bgr = cv2.imread('data/images/bus.jpg')
> im_rgb1 = im_bgr[...,::-1]
> im_rgb2 = cv2.cvtColor(im_bgr, cv2.COLOR_BGR2RGB)
> 
> print(im_bgr.data.contiguous)  # True
> print(im_rgb1.data.contiguous)  # False
> print(im_rgb2.data.contiguous)  # True
> ```

Mh ok, thats a good point.

**Comment by glenn-jocher** [comment]
@UnglvKitDe yes, it's important to ensure that the image arrays remain contiguous for performance reasons. The changes in the PR should not affect the contiguity of the arrays, but it's always good to verify that such issues don't arise with modifications to image handling. If you encounter any performance issues or have further questions, feel free to reach out. Thanks for your contribution! 🚀

**Pull request closed without merging**